core/modules/link/link.module

```diff
@@ -0,0 +1,1460 @@
+    '#description' => t('If checked, the URL field is optional and submitting a title alone will be acceptable. If the URL is omitted, the title will be displayed as plain text.'),
+  );
+
+  $title_options = array(
+    'optional' => t('Optional Title'),
+    'required' => t('Required Title'),
+    'value' => t('Static Title'),
+    'none' => t('No Title'),
+  );
+
+  $form['title'] = array(
+    '#type' => 'radios',
+    '#title' => t('Link Title'),
+    '#default_value' => isset($instance['settings']['title']) ? $instance['settings']['title'] : 'optional',
+    '#options' => $title_options,
+    '#description' => t('If the link title is optional or required, a field will be displayed to the end user. If the link title is static, the link will always use the same title. The static title value may use tokens of any other entity field as its value. Static and token-based titles may include most inline XHTML tags such as <em>strong</em>, <em>em</em>, <em>img</em>, <em>span</em>, etc.'),
+  );
+
+  $form['title_value'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Static title'),
+    '#default_value' => isset($instance['settings']['title_value']) ? $instance['settings']['title_value'] : '',
+    '#description' => t('This title will always be used if &ldquo;Static Title&rdquo; is selected above.'),
+  );
+
+  $form['title_label_use_field_label'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Use field label as the label for the title field'),
+    '#default_value' => isset($instance['settings']['title_label_use_field_label']) ? $instance['settings']['title_label_use_field_label'] : FALSE,
+    '#description' => t('If this is checked the field label will be hidden.'),
+  );
+
+  $form['title_maxlength'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Max length of title field'),
+    '#default_value' => isset($instance['settings']['title_maxlength']) ? $instance['settings']['title_maxlength'] : '128',
+    '#description' => t('Set a maximum length on the title field (applies only if Link Title is optional or required).  The maximum limit is 255 characters.'),
+    '#maxlength' => 3,
+    '#size' => 3,
+  );
+
+  if (module_exists('token')) {
+    // Add token module replacements fields.
+    $form['enable_tokens'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Allow user-entered tokens'),
+      '#default_value' => isset($instance['settings']['enable_tokens']) ? $instance['settings']['enable_tokens'] : 1,
+      '#description' => t('Checking will allow users to enter tokens in URLs and Titles on the entity edit form. This does not affect the field settings on this page.'),
+    );
+
+    $entity_info = entity_get_info($instance['entity_type']);
+    $form['tokens_help'] = array(
+      '#theme' => 'token_tree',
+      '#token_types' => array($entity_info['token type']),
+      '#global_types' => TRUE,
+      '#click_insert' => TRUE,
+      '#dialog' => TRUE,
+    );
+  }
```

Comment:
This needs to be updated for token in core. No `if` statement needed, and instead of using a token_tree element, we can just append a link to the description:

```php
  $token_link = theme('token_tree_link', array(
    'token_types' => array($entity_info['token type']),
  ));
```